miden-tx/src/errors/mod.rs

```diff
@@ -48,6 +48,7 @@ pub enum TransactionProverError {
     InvalidTransactionOutput(TransactionOutputError),
     ProvenTransactionError(ProvenTransactionError),
     TransactionHostCreationFailed(TransactionHostError),
+    DeserializationError(String),
```

Comment:
I might have missed it, but are we actually using this enum somewhere?

Reply:
We are using it in the `impl TryFrom<ProveTransactionResponse> for ProvenTransaction` implementation.

Reply:
Ah - I think we can probably just return `DeserializationError` from there (since it is internal to the crate) and remove this variant.